app/models/bookmark.rb

```diff
@@ -126,9 +126,10 @@ def invalidate_bookmark_count
   end
 
   # We index the bookmark count, so if it should change, update the pseud
-  def update_pseud_index
+  def update_pseud_and_collection_index
     return unless destroyed? || saved_change_to_id? || saved_change_to_private? || saved_change_to_hidden_by_admin?
     IndexQueue.enqueue_id(Pseud, pseud_id, :background)
+    IndexQueue.enqueue_ids(Collection, collection_ids, :background)
```

Comment:
Do we also need to reindex any parent collections here? The bookmarked item count for parent collections includes the works for subcollections. It's true the bookmarked item count won't _always_ change in these scenarios -- e.g. if two people have bookmarked Work A, and only one of those bookmarks is hidden by an admin -- but I think it will sometimes (e.g. if only one person has bookmarked Work A and then sets the bookmark to private).

(h/t ticking for [mentioning a similar issue on the CollectionItems model](https://github.com/otwcode/otwarchive/pull/3912#discussion_r528185995), which is now resolved.)

Reply:
Excellent, thanks!